pkg/collector/runner/num_workers_test.go

```diff
@@ -0,0 +1,227 @@
+package runner
+
+import (
+	"runtime"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/collector/py"
+	python "github.com/sbinet/go-python"
+)
+
+/****************** Testing configuration ****************************/
+
+var testingEfficiency = false // Run this test (should be false normally)
+var static = false            // Run with default num workers (vs dynamic)
+var prettyOutput = false      // Labelled test results
+var checkType = busyWait      // Which type of check to run (busyWait, lazyWait, or pythonCheck)
+var numIntervals = 10         // How many times to repeat each check (for the time tests)
+var memoryTest = false        // After the time tests, run the memory test
+
+/*********************************************************************/
+
+type CheckType int
+
+const (
+	busyWait    CheckType = iota
+	lazyWait    CheckType = iota
+	pythonCheck CheckType = iota // BUG: testing the python checks crashes every so often
```

Comment:
Also, what happened to the `BUG` in the comment? Do we no longer crash? If so, could we remove this comment?

Reply:
Unfortunately, we do still crash occasionally... discussed with @masci and decided that it wasn't worth it to spend more time than we already have investigating the crash since I was able to run the tests enough times successfully to still be able to collect the data we wanted